worker/src/main/java/gov/cms/ab2d/worker/processor/PatientClaimsProcessorImpl.java

```diff
@@ -151,34 +155,48 @@ private void handleException(StreamHelper helper, String data, Exception e) thro
         }
 
         final List<BundleEntryComponent> entries = eobBundle.getEntry();
-        final List<Resource> resources = extractResources(entries, patient.getDateRangesUnderContract(), attTime, map);
+        final List<Resource> resources = extractResources(request.getContractNum(), entries, patient.getDateRangesUnderContract(), attTime, map);
 
         while (eobBundle.getLink(Bundle.LINK_NEXT) != null) {
             eobBundle = bfdClient.requestNextBundleFromServer(eobBundle);
             final List<BundleEntryComponent> nextEntries = eobBundle.getEntry();
-            resources.addAll(extractResources(nextEntries, patient.getDateRangesUnderContract(), attTime, map));
+            resources.addAll(extractResources(request.getContractNum(), nextEntries, patient.getDateRangesUnderContract(), attTime, map));
         }
 
         log.debug("Bundle - Total: {} - Entries: {} ", eobBundle.getTotal(), entries.size());
         return resources;
     }
 
-    List<Resource> extractResources(List<BundleEntryComponent> entries, final List<FilterOutByDate.DateRange> dateRanges,
-                                    OffsetDateTime attTime, Map<String, ContractBeneficiaries.PatientDTO> patientsMap) {
-        if (attTime == null) {
-            return new ArrayList<>();
-        }
-        long epochMilli = attTime.toInstant().toEpochMilli();
-        Date attDate = new Date(epochMilli);
+    private Date getStartDate(String contract) {
         SimpleDateFormat sdf = new SimpleDateFormat("MM/dd/yyyy");
+
+        String dateToUse = startDate;
+        if (isContractSpecial(contract)) {
+            dateToUse = startDateSpecialContracts;
+        }
         Date date;
         try {
-            date = sdf.parse(startDate);
+            date = sdf.parse(dateToUse);
         } catch (ParseException e) {
             LocalDateTime d = LocalDateTime.of(2020, 1, 1, 0, 0, 0, 0);
             date = new Date(d.toInstant(ZoneOffset.UTC).toEpochMilli());
         }
-        final Date earliestDate = date;
+        return date;
+    }
+
+    private boolean isContractSpecial(String contract) {
+        return this.specialContracts != null && !this.specialContracts.isEmpty() && specialContracts.contains(contract);
+    }
+
+    List<Resource> extractResources(String contractNum, List<BundleEntryComponent> entries,
+                                    final List<FilterOutByDate.DateRange> dateRanges,
+                                    OffsetDateTime attTime, Map<String, ContractBeneficiaries.PatientDTO> patientsMap) {
```

Comment:
I noticed a blending of the deprecated java.util.Date along with the newer java.time classes.

Given that the usage of Date is transitory here, I don't see a concrete issue but do suggest avoidance of java.util.Date.

No actual changes are necessary at this time, this is only a suggestion.

Reply:
It's used by the FHIR search API so it makes its way back into the code on occasion to prevent tons of back and forth transformations but I definitely agree with the sentiment.